ns/company.go

```diff
@@ -0,0 +1,39 @@
+package ns
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+)
+
+// CompanyRequest The structure of the request will be made to the company endpoint.
+type CompanyRequest struct {
+	Credentials *Credentials `json:"credentials"`
+}
+
+// CompanyService operates over company requests.
+type CompanyService service
+
+// Companies returns all companies.
+func (cs *CompanyService) Companies(crq *CompanyRequest) (clr []*CompanyListResponse, err error) {
```

Comment:
You are still returning an array.
```suggestion
func (cs *CompanyService) Companies(crq *CompanyRequest) (clr *CompanyListResponse, err error) {
```